File: src/main/java/no/nav/service/pensjon/regler/to/BeregnInstitusjonsoppholdRequest.java
package no.nav.service.pensjon.regler.to;

import java.util.Date;
import java.util.Vector;

import no.nav.domain.pensjon.regler.beregning.Beregning;
import no.nav.domain.pensjon.regler.beregning2011.AbstraktBeregningsResultat;
import no.nav.domain.pensjon.regler.beregning2011.AfpLivsvarig;
import no.nav.domain.pensjon.regler.beregning2011.BeregningsresultatUforetrygd;
import no.nav.domain.pensjon.regler.beregning2011.SisteAldersberegning2011;
import no.nav.domain.pensjon.regler.grunnlag.ForholdstallUtvalg;
import no.nav.domain.pensjon.regler.krav.Kravhode;
import no.nav.domain.pensjon.regler.vedtak.VilkarsVedtak;

/**
 * @author PREG
 * @author Stian Kroknes (Accenture) PENPORT-1055
 * @author Nabil Safadi (Decisive) - PK-8518
 */
public class BeregnInstitusjonsoppholdRequest extends ServiceRequest {
	private Date fom;
	
	// bruker1 beregning1967
	private Beregning bruker1Beregning;
	private Kravhode bruker1Kravhode; 
	private Vector<VilkarsVedtak> bruker1VilkarsvedtakListe = new Vector<VilkarsVedtak>();
	
	// bruker1  beregningsResultat(dekker 2011,2016 og 2025)
	private AbstraktBeregningsResultat bruker1BeregningsResultat;
	private Kravhode bruker1Kravhode2011;
	private Vector<VilkarsVedtak> bruker1VilkarsvedtakListe2011 = new Vector<VilkarsVedtak>();
	private ForholdstallUtvalg bruker1ForholdstallUtvalg;
	private SisteAldersberegning2011 bruker1SisteAldersberegning2011;
    /* Informasjon om uføretrygdberegning for bruker1. Kun påkrevd dersom bruker1 har uføretrygd. */
    private BeregningsresultatUforetrygd bruker1BeregningsresultatUforetrygd;

    /* Informasjon om afpPrivatberegning for bruker1. Kun påkrevd dersom bruker1 har AfpPrivat. */
	private AfpLivsvarig bruker1afpLivsvarig;
	
	// bruker2 beregning1967
	private Beregning bruker2Beregning;
	private Kravhode bruker2Kravhode;
	private Vector<VilkarsVedtak> bruker2VilkarsvedtakListe = new Vector<VilkarsVedtak>();
	
	// bruker2  beregningsResultat(dekker 2011,2016 og 2025)
	private AbstraktBeregningsResultat bruker2BeregningsResultat;
	private Kravhode bruker2Kravhode2011;
	private Vector<VilkarsVedtak> bruker2VilkarsvedtakListe2011 = new Vector<VilkarsVedtak>();
	private ForholdstallUtvalg bruker2ForholdstallUtvalg;
	private SisteAldersberegning2011 bruker2SisteAldersberegning2011;

	/* Informasjon om afpPrivatberegning for bruker2. Kun påkrevd dersom bruker2 har AfpPrivat. */
	private AfpLivsvarig bruker2afpLivsvarig;
	

}

File: src/main/kotlin/no/nav/pensjon/regler/to/BeregnInstitusjonsoppholdRequest.kt
package no.nav.pensjon.regler.to

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import no.nav.pensjon.regler.domain.beregning.Beregning
import no.nav.pensjon.regler.domain.beregning2011.AbstraktBeregningsResultat
import no.nav.pensjon.regler.domain.beregning2011.AfpPrivatLivsvarig
import no.nav.pensjon.regler.domain.beregning2011.BeregningsresultatUforetrygd
import no.nav.pensjon.regler.domain.beregning2011.SisteAldersberegning2011
import no.nav.pensjon.regler.domain.krav.Kravhode
import no.nav.pensjon.regler.domain.vedtak.VilkarsVedtak
import java.time.LocalDate
import java.util.*

class BeregnInstitusjonsoppholdRequest : ServiceRequest() {
    var virk: LocalDate? = null

    @Deprecated(
        "Bruk virk i stedet",
        ReplaceWith("virk"),
        level = DeprecationLevel.WARNING)
    @set:JsonProperty("fom")
    @get:JsonIgnore
    var fom: Any? // Any fordi gammel fom kunne være Long (Date)
        get() = virk
        set(value) { /* Verdien mappes via MixIn i mottaker-systemet */ }

    @Deprecated(
        "Bruk virk i stedet",
        ReplaceWith("virk"),
        level = DeprecationLevel.WARNING
    )
    @set:JsonProperty("fomLd")
    @get:JsonIgnore
    var fomLd: LocalDate?
        get() = virk
        set(value) {
            virk = value
        }

    // bruker1 beregning1967
    var bruker1Beregning: Beregning? = null
    var bruker1Kravhode: Kravhode? = null
    var bruker1VilkarsvedtakListe = Vector<VilkarsVedtak>()

    // bruker1  beregningsResultat(dekker 2011,2016 og 2025)
    var bruker1BeregningsResultat: AbstraktBeregningsResultat? = null
    var bruker1Kravhode2011: Kravhode? = null
    var bruker1VilkarsvedtakListe2011 = Vector<VilkarsVedtak>()
    var bruker1SisteAldersberegning2011: SisteAldersberegning2011? = null

    /* Informasjon om uføretrygdberegning for bruker1. Kun påkrevd dersom bruker1 har uføretrygd. */
    var bruker1BeregningsresultatUforetrygd: BeregningsresultatUforetrygd? = null

    /* Informasjon om afpPrivatberegning for bruker1. Kun påkrevd dersom bruker1 har AfpPrivat. */
    var bruker1afpPrivatLivsvarig: AfpPrivatLivsvarig? = null

    // bruker2 beregning1967
    var bruker2Beregning: Beregning? = null
    var bruker2Kravhode: Kravhode? = null
    var bruker2VilkarsvedtakListe = Vector<VilkarsVedtak>()

    // bruker2  beregningsResultat(dekker 2011,2016 og 2025)
    var bruker2BeregningsResultat: AbstraktBeregningsResultat? = null
    var bruker2Kravhode2011: Kravhode? = null
    var bruker2VilkarsvedtakListe2011 = Vector<VilkarsVedtak>()
    var bruker2SisteAldersberegning2011: SisteAldersberegning2011? = null

    /* Informasjon om afpPrivatberegning for bruker2. Kun påkrevd dersom bruker2 har AfpPrivat. */
    var bruker2AfpPrivatLivsvarig: AfpPrivatLivsvarig? = null
}